ocaml/xapi/xapi_pool.ml

```diff
@@ -3228,7 +3228,50 @@ let alert_failed_login_attempts () =
             ~body:stats
       )
 
+let perform ~local_fn ~__context ~host op =
+  let rpc' context hostname (task_opt : API.ref_task option) xml =
+    let open Xmlrpc_client in
+    (* use certificate checking unconditionally: Some Stunnel.pool *)
+    let transport =
+      SSL
+        ( SSL.make ~verify_cert:(Some Stunnel.pool)
+            ?task_id:(Option.map Ref.string_of task_opt)
+            ()
+        , hostname
+        , !Constants.https_port
+        )
+    in
+    let http =
+      xmlrpc ?task_id:(Option.map Ref.string_of task_opt) ~version:"1.0" "/"
+    in
+    XMLRPC_protocol.rpc ~srcstr:"xapi" ~dststr:"dst_xapi" ~transport ~http xml
+  in
```

Comment:
Instead of redefining the rpc function, we could add a optional `force_cert_check` parameter to `message_forwarding.ml:remote_rpc_retry` and use that here

Reply:
I am not sure I agree. This requires quite a bit of parameter threading in a code base that is complex already. Here we do one thing locally and that is it. I've cleaned up the code a bit.